openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py

```diff
@@ -0,0 +1,157 @@
+
+from openbb_charting import charting_router
+from openbb_charting.core.to_chart import ChartIndicators, OpenBBFigure, to_chart
+from openbb_charting.utils.helpers import get_charting_functions
+
+ext = Extension(name="charting")
+
+
+@ext.obbject_accessor
+class Charting:
+    """
+    Charting extension.
+
+    Methods
+    -------
+    show
+        Display chart and save it to the OBBject.
+    to_chart
+        Returns the plotly json representation of the chart.
+    functions
+        Returns a list of Platform commands with charting functions.
+    indicators
+        Returns a list of the available indicators to use with the `to_chart` method.
+    """
+
+    def __init__(self, obbject):
+        """Initialize Charting extension."""
+        # pylint: disable=import-outside-toplevel
+        from openbb_core.app.model.charts.charting_settings import ChartingSettings
+        from openbb_core.app.model.obbject import OBBject
+
+        self._obbject: OBBject = obbject
+        self._charting_settings = ChartingSettings(
+            user_settings=self._obbject._user_settings,
+            system_settings=self._obbject._system_settings,
+        )
+        self._handle_backend()
+
+    @classmethod
+    def indicators(cls):
+        """Returns a list of the available indicators."""
+        return ChartIndicators.get_available_indicators()
+
+    @classmethod
+    def functions(cls):
+        """Returns a list of the available functions."""
+        return get_charting_functions()
+
+    def _handle_backend(self):
+        # pylint: disable=import-outside-toplevel
+        from openbb_charting.core.backend import create_backend, get_backend
+
+        create_backend(self._charting_settings)
+        get_backend().start(debug=self._charting_settings.debug_mode)
+
+    @staticmethod
+    def _get_chart_function(route: str) -> Callable:
+        """Given a route, it returns the chart function. The module must contain the given route."""
+        adjusted_route = route.replace("/", "_")[1:]
+        return getattr(charting_router, adjusted_route)
+
+    def show(self, render: bool = True, **kwargs):
+        """Display chart and save it to the OBBject."""
+        charting_function = self._get_chart_function(
+            self._obbject._route  # pylint: disable=protected-access
+        )
+        kwargs["obbject_item"] = self._obbject.results
+        kwargs["charting_settings"] = self._charting_settings
+        kwargs["standard_params"] = (
+            self._obbject._standard_params.__dict__  # pylint: disable=protected-access
+        )
+
+        fig, content = charting_function(**kwargs)
+        self._obbject.chart = Chart(
+            fig=fig, content=content, format=charting_router.CHART_FORMAT
+        )
+        if render:
+            fig.show()
+
+    def to_chart(
+        self,
+        indicators: Optional[Union[ChartIndicators, Dict[str, Dict[str, Any]]]] = None,
+        symbol: str = "",
+        candles: bool = True,
```

Comment:
I also find myself looking for a way to make this useful for other types of data, or where the type of chart desired is a different format.  Like the entry point for data, but then a way to define X/Y based on a list of fields within Data.

Reply:
I like the idea but for now, I think we need to keep adding changes, PR is too big already! 
IMO, the primary focus regarding the charting extension moving forward should be to actually implement more views.
@deeleeramone would you mind documenting this in our usual Jira ticket so that we don't lose this insight?